src/from_substrait.cpp

```diff
@@ -700,6 +700,10 @@ shared_ptr<Relation> SubstraitToDuckDB::GetValuesExpression(const google::protob
 		}
 		expressions.emplace_back(std::move(expression_row));
 	}
+	if (expressions.size() == 1 && expressions[0].empty()) {
+		// add a dummy row, as DuckDB does not support zero column ValueRelation
```

Comment:
Does adding a row change what the result would have been?  For example does a select * now return a row when it wouldn't have before?  This transformation is only safe when the result isn't consumed (no field references by its consumer or theirs until the column disappears).

Reply:
Sorry the comment was wrong, it should be  `dummy column`.

I am adding this dummy column when there is only one row and no columns in the virtual table. 
Yes it is safe only when there are no column references. If there are any column references then the plan is invalid to start with.